File: week2/atakantekatan/stack.py
class Node:
    def __init__(self, val):
        self.value = val
        self.next = None
        
class Stack:
    def __init__(self):
        self.head = Node("head")
        self.len = 0
    
    def size(self):
        return self.len
    
    def isEmpty(self):
        return True if self.len == 0 else False
    
    def push(self,val):
        newest = Node(val)
        newest.next = self.head.next
        self.head.next = newest
        self.len += 1
        
    def pop(self):
        if self.len == 0:
            raise Exception("Can't pop from empty stack")
        latest = self.head.next
        self.head.next = self.head.next.next
        self.len -= 1
        return latest.value
        
    def top(self):
        if self.len == 0:
            raise Exception("Stack has no elements in it")
        return self.head.next.value
        
stack = Stack()
stack.push(123)

File: week2/Can-Yavuz/Stack.cs
public class MyStack
{
    private ListNode top;

    public MyStack()
    {
        top = null;
    }

    public void Push(int x)
    {
        ListNode newNode = new ListNode(x);
        newNode.next = top;
        top = newNode;
    }

    public int Pop()
    {
        if (top == null)
            throw new InvalidOperationException("Stack is empty.");

        int value = top.val;
        top = top.next;
        return value;
    }

    public int Peek()
    {
        if (top == null)
            throw new InvalidOperationException("Stack is empty.");

        return top.val;
    }

    public bool IsEmpty()
    {
        return top == null;
    }
}
